packages/webhooks/tasks/deleteOldWebhookPayloads.js

```diff
@@ -0,0 +1,57 @@
+const dayjs = require('dayjs')
+
+const { getLogger } = require('@open-condo/keystone/logging')
+const { itemsQuery, getSchemaCtx } = require('@open-condo/keystone/schema')
+const { WEBHOOK_PAYLOAD_RETENTION_DAYS } = require('@open-condo/webhooks/constants')
+const { WebhookPayload } = require('@open-condo/webhooks/schema/utils/serverSchema')
+
+const logger = getLogger()
+
+const BATCH_SIZE = 100
+
+/**
+ * Hard deletes old WebhookPayload records from the database.
+ * Deletes records not updated for more than WEBHOOK_PAYLOAD_RETENTION_DAYS (default: 42 days).
+ * This helps keep the database clean and prevents unbounded growth of stored data.
+ */
+async function deleteOldWebhookPayloads () {
+    const { keystone } = getSchemaCtx('WebhookPayload')
+    const context = await keystone.createContext({ skipAccessControl: true })
+    const cutoffDate = dayjs().subtract(WEBHOOK_PAYLOAD_RETENTION_DAYS, 'day').toISOString()
+
+    logger.info({ msg: 'Starting cleanup of old webhook payloads', data: { cutoffDate, retentionDays: WEBHOOK_PAYLOAD_RETENTION_DAYS } })
+
+    let totalDeleted = 0
+
+    let hasMore = true
+    while (hasMore) {
```

Comment:
I would say, that there will be good to see something like "maxIterations" limiter. 
So to be sure that this task won't run hours if there is a lot of data to remove

Reply:
All will be ok. However, if the task cannot remove data for hours, we need to refactor it.

Reply:
We need to limit task execution time, so consider adding upper limit to amount of iterations. 
Same for `retryFailed...`